bin/anthology/utils.py

```diff
@@ -120,6 +120,31 @@ def test_url(url):
     return test_url_code(url).status_code == requests.codes.ok
 
 
+def download_url(remote_url, local_path):
+    """
+    Downloads a remote URL to a local path. Can handle cookies, e.g., those
+    used downloading PDFs from MIT Press (TACL, CL).
+    """
+    if remote_url.startswith("http"):
+        import ssl
+        import urllib.request
+
+        try:
+            cookieProcessor = urllib.request.HTTPCookieProcessor()
+            opener = urllib.request.build_opener(cookieProcessor)
+            request = urllib.request.Request(
+                remote_url, headers={'User-Agent': 'Mozilla/5.0'}
+            )
+            with opener.open(request, timeout=1000) as url, open(
+                local_path, mode="wb"
+            ) as input_file_fh:
+                input_file_fh.write(url.read())
+        except:
```

Comment:
This method returns the path on success (which is not altered my the method) and False on error.  Nobody is interested in the path (caller has the path already) and it should return True on success instead.  The code now depends on non-empty strings being true, which is a brittle: `if download_url(...):` works, but `if download_url(...) == True:` does not.

I would rather throw an exception than return true / false to not have silent errors when the method is not called in an if clause, but that might be my personal preference.

Reply:
Thanks, this is good feedback. Returning `True` is helpful, I think, but I removed the Exception catching so that will get passed up the call stack.

Reply:
See if you can tolerate the version I checked in...

Reply:
I can tolerate that :-) but now it always returns `True` and never false, so you could get rid of the return true altogether.  Exception handling kicks in on exception anyway, so the if clause in `if (retrieve_url(...))` is not needed and the `return True` is also not needed.